src/networkd.c

```diff
@@ -1499,6 +1499,15 @@ _netplan_networkd_write_wait_online(const NetplanState* np_state, const char* ro
         _netplan_g_string_free_to_file(content, rootdir, override, NULL);
         g_hash_table_destroy(non_optional_interfaces);
         return TRUE;
+    } else {
+        // Systemd's /lib/systemd/system/systemd-networkd.service installs an
+        // enablement link for systemd-networkd-wait-online.service in its
+        // [Install] section via "Also=systemd-networkd-wait-online.service".
+        // We want to avoid it being called without --interface arguments, to
+        // avoid a boot delay, when waiting for configured, but optional interfaces.
+        GString* content_noop = g_string_new("[Service]\nExecStart=\n"
+                                             "ExecStart=/bin/true\n");
```

Comment:
note: I'm not so sure we want this behavior by default, at least not on non-Ubuntu systems.